addons/website/models/website.py

```diff
@@ -1785,60 +1784,18 @@ def _trigram_enumerate_words(self, search_details, search, limit):
             domain = search_detail['base_domain'].copy()
             fields = set(fields).intersection(model._fields)
 
-            unaccent = get_unaccent_wrapper(self.env.cr)
+            query = Query(self.env.cr, model._table, model._table_query)
 
-            # Specific handling for fields being actually part of another model
-            # through the `inherits` mechanism.
-            # It gets the list of fields requested to search upon and that are
-            # actually not part of the requested model itself but part of a
-            # `inherits` model:
-            #     {
-            #       'name': {
-            #           'table': 'ir_ui_view',
-            #           'fname': 'view_id',
-            #       },
-            #       'url': {
-            #           'table': 'ir_ui_view',
-            #           'fname': 'view_id',
-            #       },
-            #       'another_field': {
-            #           'table': 'another_table',
-            #           'fname': 'record_id',
-            #       },
-            #     }
-            inherits_fields = {
-                inherits_model_fname: {
-                    'table': self.env[inherits_model_name]._table,
-                    'fname': inherits_field_name,
-                }
-                for inherits_model_name, inherits_field_name in model._inherits.items()
-                for inherits_model_fname in self.env[inherits_model_name]._fields.keys()
-                if inherits_model_fname in fields
-            }
-            similarities = []
-            for field in fields:
-                # Field might belong to another model (`inherits` mechanism)
-                table = inherits_fields[field]['table'] if field in inherits_fields else model._table
-                similarities.append(
-                    sql.SQL("word_similarity({search}, {field})").format(
-                        search=unaccent(sql.Placeholder('search')),
-                        field=unaccent(sql.SQL("{table}.{field}").format(
-                            table=sql.Identifier(table),
-                            field=sql.Identifier(field)
-                        )) if not model._fields[field].translate else
-                        unaccent(sql.SQL("COALESCE({table}.{field}->>{lang}, {table}.{field}->>'en_US')").format(
-                            table=sql.Identifier(table),
-                            field=sql.Identifier(field),
-                            lang=sql.Literal(lang)
```

Comment:
Isn't this PR about some change on `lang` check?
Why can this whole "search on inherit**s** fields too" now be removed?
Does Odoo tool's `SQL` take care of that automatically?
Don't forget to comment the change in website, we touch a lot those lines, it will help figuring things when blaming :+1:

Reply:
I quick answer here: (I will add these info to commit message for website)
This task wants to remove `lang` check, and use `_field_to_sql` to inject `->>%(lang)s` in a safe way.
Benifiting from `_field_to_sql`, which can also handel table `JOIN` for inherited fields, the old hack is not necessary anymore.